mmcv/transform/wrappers.py

```diff
@@ -0,0 +1,465 @@
+        data items according to the input_mapping.
+
+        Args:
+            data (dict): The original input data
+            input_mapping (dict): The input key mapping. See the document of
+                mmcv.transforms.wrappers.Remap` for details.
+
+        Returns:
+            dict: The input data with remapped keys. This will be the actual
+                input of the wrapped pipeline.
+        """
+
+        def _remap(data, m):
+            if isinstance(m, dict):
+                # m is a dict {inner_key:outer_key, ...}
+                return {k_in: _remap(data, k_out) for k_in, k_out in m.items()}
+            if isinstance(m, (tuple, list)):
+                # m is a list or tuple [outer_key1, outer_key2, ...]
+                # This is the case when we collect items from the original
+                # data to form a list or tuple to feed to the wrapped
+                # transforms.
+                return m.__class__(_remap(data, e) for e in m)
+
+            # m is an outer_key
+            if self.strict:
+                return data.get(m)
+            else:
+                return data.get(m, NotInResults)
+
+        collected = _remap(data, input_mapping)
+        collected = {
+            k: v
+            for k, v in collected.items() if v is not NotInResults
+        }
+
+        # Retain unmapped items
+        inputs = data.copy()
+        inputs.update(collected)
+
+        return inputs
+
+    def remap_output(self, data: Dict, output_mapping: Dict) -> Dict[str, Any]:
+        """Remap outputs from the wrapped transforms by gathering and renaming
+        data items according to the output_mapping.
+
+        Args:
+            data (dict): The output of the wrapped pipeline.
+            output_mapping (dict): The output key mapping. See the document of
+                `mmcv.transforms.wrappers.Remap` for details.
+
+        Returns:
+            dict: The output with remapped keys.
+        """
+
+        def _remap(data, m):
+            if isinstance(m, dict):
+                assert isinstance(data, dict)
+                results = {}
+                for k_in, k_out in m.items():
+                    assert k_in in data
+                    results.update(_remap(data[k_in], k_out))
+                return results
+            if isinstance(m, (list, tuple)):
+                assert isinstance(data, (list, tuple))
+                assert len(data) == len(m)
+                results = {}
+                for m_i, d_i in zip(m, data):
+                    results.update(_remap(d_i, m_i))
+                return results
+
+            if data is NotInResults:
+                raise ValueError(
+                    f'Attempt to assign `NotInResults` to output key {m}.'
+                    '`NotInResults` just serves as a placeholder for missing '
+                    'keys in non-strict input mapping. It should not be '
+                    'assigned to any output.')
```

Comment:
Is this case tested?

Reply:
On second thought this part may be unnecessary. `NotInResults` should never appear in the output dict. I will remove this part.